ARCHITECTURE.md

```diff
@@ -0,0 +1,140 @@
+# CI Operator Architecture
+
+The CI Operator has a number of foundational principles:
+
+ - don't do work twice
+ - do the least amount of work needed
+ - hide complexity from the end user
+
+In order to achieve this, the CI Operator requires configuration to understand
+the build process for each component as well as every output container image.
+This document will overview the workflow that the CI Operator uses to build
+components and structure tests.
+
+Every invocation of `ci-operator` will create a workspace to isolate test
+execution, seed it with build inputs and the published component images from
+the OpenShift release that the test is a part of, then schedule test workflows
```

Comment:
'...the test is a part of...' - does this make sense? the test is probably not part of the release, a tested component is?